### PR TITLE
Implements a negate method for Amount

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/java/sirius/kernel/commons/Amount.java [hunk 1]
```diff
@@ -625,6 +625,16 @@ public Amount max(@Nullable Amount other) {
         return this.value.compareTo(other.value) > 0 ? this : other;
     }
 
+    /**
+     * Negates <tt>this</tt> amount and returns the new amount.
+     *
+     * @return an <tt>Amount</tt> representing the negated <tt>Amount</tt>. If <tt>this</tt> is empty, <tt>NOTHING</tt>
+     * is returned.
+     */
+    public Amount negate() {
```
[the remark this answers is not in the document]
Comment (jmuscireum) [reply src/main/java/sirius/kernel/commons/Amount.java]: done

File: src/test/java/sirius/kernel/commons/AmountSpec.groovy [hunk 2]
```diff
@@ -148,4 +148,26 @@ class AmountSpec extends BaseSpecification {
...
 
+    def "basic artithmetic works"() {
+        expect:
+        Amount.of(4.2).add(Amount.of(42)) == Amount.of(46.2)
+        Amount.of(4.2).subtract(Amount.of(42)) == Amount.of(-37.8)
+        Amount.of(42).subtract(Amount.of(4.2)) == Amount.of(37.8)
+        Amount.of(4.2).times(Amount.of(42)) == Amount.of(176.4)
+        Amount.of(4.2).divideBy(Amount.of(42)) == Amount.of(0.1)
+        Amount.of(42).divideBy(Amount.of(4.2)) == Amount.TEN
+        Amount.of(4.2).negate() == Amount.of(-4.2)
```
Comment (andyHa) [inline src/test/java/sirius/kernel/commons/AmountSpec.groovy]: test edge cases like 0 and nothing to ensure correctness across refactorings

File: src/test/java/sirius/kernel/commons/AmountSpec.groovy [hunk 3]
```diff
@@ -148,4 +148,26 @@ class AmountSpec extends BaseSpecification {
...
+        Amount.of(-4.2).negate() == Amount.of(4.2)
+    }
+
+    def "percent artithmetic works"() {
+        expect:
+        Amount.of(4.2).increasePercent(Amount.TEN) == Amount.of(4.62)
```
Comment (meg1502) [inline src/test/java/sirius/kernel/commons/AmountSpec.groovy]: use data tables to make these tests more tidy

http://spockframework.org/spock/docs/1.0/data_driven_testing.html
